### PR TITLE
Fix GLM5 stale tests: align with current lite API

### PR DESCRIPTION
**Pull request by ISEEKYAN**
## Summary

- Fix `test_glm5_lite_static.py` and `test_glm5_lite_cp_smoke.py` to match the current Megatron Lite API (checkpoint load/save, labeled-output interface).
- Previously these tests referenced symbols and call signatures that no longer exist in the codebase, causing CI failures on any branch that includes the GLM5 model.

## Changes

- **`test_glm5_lite_static.py`**: Updated import paths and API calls to align with current `lite` module structure.
- **`test_glm5_lite_cp_smoke.py`**: Updated to use the labeled-output API (CP smoke path).

## Validation

- Static tests: Slurm job 13039052 — 4 targets PASS
- CP smoke: Slurm job 13052759 — COMPLETED rc=0, non-skip

## Notes

These were pre-existing stale tests in `is/main` (not introduced by any split). This PR is a prerequisite for the model PRs in the `dev-mlite-5/6/7` series to have clean, passing tests.

### CODE REVIEW

**Comment by ISEEKYAN** [comment]
## Verification

Ran the full GLM5 static suite on a GPU node (the native GLM-5 path requires Transformer Engine RMSNorm, which is CUDA-only):

```
experimental/lite/tests/unit/model/test_glm5_lite_static.py
17 passed, 0 failed, 0 skipped   (cuda_available=True)
```

### What changed
- **MTP assertions**: split into a no-labels inference call (asserts `mtp_logits`) and a labeled training call (asserts `mtp_loss`) — the labeled path returns the MTP loss, not per-head logits.
- **Parallel-scope test**: assert CP-only and PP/VPP validate cleanly (they are supported) and cover the TP>1 / ETP>1 rejection.
- **Forward-running tests**: place model/attention/inputs on CUDA with bf16 params (matching the CP-smoke convention) and skip cleanly when no GPU is present. Export/state-name/checkpoint tests stay on CPU.
- **Shape assertions** aligned to the real output convention: `hidden_states`/`mtp_hidden_states` are `(seq, batch, hidden)`; `logits`/`mtp_logits` are transposed back to `(batch, seq, *)`.

Only test files are touched (`test_glm5_lite_static.py`, `test_glm5_lite_cp_smoke.py`); no implementation/checkpoint changes.
